pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -943,6 +1104,43 @@ func (vm *VirtualMachine) GetVirtualHardDiskByPath(path string) (vhd *disk.Virtu
 	return
 }
 
+func (vm *VirtualMachine) GetAttachedVirtualHardDisks() (vhdPaths []string, err error) {
+	col, err := vm.GetVirtualHardDisks()
+	if err != nil {
+		return
+	}
+	defer col.Close()
+
+	for _, inst := range col {
+		tmpvhd, err1 := disk.NewVirtualHardDisk(inst.WmiInstance)
```

Comment:
I think these NewVirtualHardDisk should be close too.

Reply:
Although the API contains the word `New`, `tmpvhd` still refers to the same WMI instance which is passed into the function. This instance is closed via line `1112` at the end of the function call.

We do not need to close `tmpvhd`